lib/mix/lib/mix/utils.ex

```diff
@@ -333,25 +379,27 @@ defmodule Mix.Utils do
   The callback will be invoked for each node and it
   must return a `{printed, children}` tuple.
 
-  If `path` is `-`, prints the output to standard output.
+  If the `:output` option is `-` then prints to standard output.
   """
   @spec write_dot_graph!(
-          Path.t(),
           String.t(),
           [node],
           (node -> {formatted_node, [node]}),
           keyword
-        ) :: :ok
+        ) :: Path.t()
+        when node: term()
+
+  @spec write_dot_graph!(
+          String.t(),
+          [node],
+          (node -> {formatted_node, [node]}),
+          keyword
+        ) :: Path.t()
         when node: term()
-  def write_dot_graph!(path, title, nodes, callback, _opts \\ []) do
+  def write_dot_graph!(title, nodes, callback, opts \\ []) do
     {dot, _} = build_dot_graph(make_ref(), nodes, MapSet.new(), callback)
     contents = ["digraph ", quoted(title), " {\n", dot, "}\n"]
-
-    if path == "-" do
-      IO.write(contents)
-    else
-      File.write!(path, contents)
-    end
+    write_according_to_opts!(opts, "xref_graph.dot", contents)
```

Comment:
This is a generic utils file, it should not have knowledge of "xref_graph.dot". The filename has to be an argument.

Reply:
ack. will fix.